### PR TITLE
feat: added generic thank you page for challenges from contentful

### PR DESCRIPTION
**Pull request by eddieferrer**
ACK-614

Screenshot:
Screen Shot 2023-06-05 at 10 48 15 AM

### CODE REVIEW

**Comment by eddieferrer** [comment]
Should we add a promise rejection is the contentful query doesnt return any content? Maybe send them to the normal thanks page?

File: src/pages/Thanks/ThanksPageChallenge.vue [hunk 1]
```diff
@@ -0,0 +1,369 @@
...
+				variables: {
+					userId: data?.my?.userAccount?.id?.toString(),
+				}
+			});
+		} catch (e) {
+			logReadQueryError(e,
+				`Thanks challenge page achievements readQuery failed: (transaction_id: ${transactionId})`);
+		}
+		this.allAchievements = achievementData?.userAchievementProgress?.achievementProgress ?? [];
+		console.log('allAchievements', this.allAchievements);
```
Comment (dyersituations) [inline src/pages/Thanks/ThanksPageChallenge.vue:348]: Remove?

File: src/pages/Checkout/CheckoutPage.vue [hunk 2]
```diff
@@ -558,31 +552,16 @@ export default {
...
-			// end game code
-		}
+		// Fetch Challenge Status
+		// If a loan in basket makes progress towards an active challenge,
+		// set query param to redirect to special thank you page
+		achievementsQuery(this.apollo, this.loanIdsInBasket)
+			.then(({ data }) => {
+				const checkoutMilestoneProgresses = data?.achievementMilestonesForCheckout?.checkoutMilestoneProgresses;
+				const challengeProgressed = achievementProgression(checkoutMilestoneProgresses);
+				this.challengeRedirectQueryParam = challengeProgressed ? `&challenge=${challengeProgressed}` : '';
```
[the remark this answers is not in the document]
Comment (eddieferrer) [reply src/pages/Checkout/CheckoutPage.vue:562]: This is setting the query string. Not sure what you mean

**Comment by eddieferrer** [comment]
> > Should we add a promise rejection is the contentful query doesnt return any content? Maybe send them to the normal thanks page?
> 
> Probably makes sense. Do we expect this to happen?

I don't really expect this to happen, I'm not even sure how it would happen, except for a user typing in an incorrect url.

I have already added all the previous challenges on contentful, so even those urls will work. It also doesn't really have any terrible consequences. If a user where to go to /checkout/thanks/foo-bar all they will see is a blank page since there won't be any data from contentful, there won't be a server error or anything

**Comment by dyersituations** [comment]
> there won't be a server error or anything

Thanks for the context, sounds like it's good as is then.

File: src/pages/Checkout/CheckoutPage.vue (hunk 2 above)
[the remark this answers is not in the document]
Comment (eddieferrer) [reply src/pages/Checkout/CheckoutPage.vue:562]: I see what you mean. Good catch! let me check

[the remark this answers is not in the document]
Comment (eddieferrer) [reply src/pages/Checkout/CheckoutPage.vue:562]: Yes this is fine, the transaction id is always passed as the first query param: `src/plugins/checkout-utils-mixin.js:179~180`

**Review by mcstover — APPROVED** [review]

**Review by dyersituations — APPROVED** [review]

**Pull request merged**